src/partisan_hiscamp_membership_strategy.erl

```diff
@@ -0,0 +1,253 @@
+        true ->
+            lager:info("~p: Issuing remove_subscription for node ~p.", [node(), Node]);
+        false ->
+            ok
+    end,
+    Membership = sets:del_element(Node, Membership0),
+    MembershipList0 = membership_list(State0),
+%% Gossip
+    Message = {remove_subscription, Node},
+    OutgoingMessages = lists:map(fun(Peer) -> {Peer, {protocol, Message}} end, MembershipList0),
+    State = State0#hiScamp{membership=Membership},
+    MembershipList = membership_list(State),
+    {ok, MembershipList, OutgoingMessages, State}.
+
+periodic(#hiScamp{last_message_time = LastMessageTime} = State) ->
+    SourceNode = myself(),
+    MembershipList = membership_list(State),
+    OutgoingPingMessages = lists:map(fun(Peer) ->
+                                             {Peer, {protocol, {ping, SourceNode}}}
+    end, MembershipList),
+    Difference = case LastMessageTime of
+                     undefined ->
+                         0;
+                     _ ->
+                         CurrentTime = erlang:timestamp(),
+                         timer:now_diff(CurrentTime, LastMessageTime)
+    end,
+    OutgoingSubscriptionMessages = case Difference > (?PERIODIC_INTERVAL * ?SCAMP_MESSAGE_WINDOW) of
+        true ->
+            case partisan_config: get(tracing, ?TRACING) of
+                true ->
+                    lager:info("~p: Node is possibly isolated.", [node()]);
+                false ->
+                    ok
+            end,
+            Myself = myself(),
+            lists:map(fun(N) ->
+                case partisan_config:get(tracing, ?TRACING) of
+                    true ->
+                        lager:info("~p: Forwarding add'l subs for ~p to node: ~p", [node(), Myself, N]);
+                    false ->
+                        ok
+                end,
+                {N, {protocol, {forward_subscription, Myself}}}
+            end, select_random_sublist(State, 1));
+        false ->
+            []
+    end,
+    {ok, MemberhsipList, OutgoingSubscriptionMessages + OutgoingPingMessages, State}.
+
+handle_message(#hiScamp{membership=Membership0}=State0, {remove_subscription, Node}) ->
+    lager:info("~p: Recieved remove_subscription for node ~p.", [node(), Node]),
+    MembershipList0 = membership_list(state0),
+    case sets:is_element(Node, Membership0) of 
+        true ->
+            Membership = sets:del_element(Membership0, Node),
+            Message = {remove_subscription, Node},
+            OutgoingMessages = lists:map(fun(Peer) -> {Peer, {protocol, Message}} end, MembershipList0),
+            State = State0#hiScamp{membership=Membership},
+            MembershipList = membership_list(state),
+            {ok, MembershipList, OutgoingMessages, State};
+        false ->
+            OutgoingMessages = [],
+            {ok, MembershipList0, OutgoingMessages, State0}
+    end;
+
+handle_message(#hiScamp{membership=Membership0}=State0, {forward_subscription, Node}) ->
```

Comment:
When a node receives a subscription request that's been forwarded (as, part of the Scamp protocol) it's altering the membership but not altering the levels... is this correct?

Reply:
No, I don't think this is correct. I should be changing the levels. First, can you tell me what's happening in this line:
OutgoingMessages = lists:map(fun(Peer) -> {Peer, {protocol, Message}} end, MembershipList0),

Reply:
That's generating a list of messages to be sent to the nodes in ```MembershipList0```.